### PR TITLE
Protein viewer TypeScript

### PR DESCRIPTION
**Pull request by gustavo-salazar**
This PR migrates the component called ProtVista into Typescript.

Given the multiple changes involved in it, I decided to migrate into a new component called ProteinViewer and split it into subcomponents.

Most components are now written as functional components with hooks.

The tooltips in the protein viewer are now using `floating-ui`. Future work will be to replace any use of `tippy` or `popper` into this lib, to unify all the tooltips.

### CODE REVIEW

File: src/utils/hooks/useInterval.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+
+type Fun = () => void;
+export const useInterval = (callback: Fun, delay: number) => {
+  const savedCallback = React.useRef<Fun>();
+  const [id, setId] = React.useState<NodeJS.Timer  |undefined>(undefined);
```
Comment (matthiasblum) [inline src/utils/hooks/useInterval.ts:6]:
```suggestion
  const [id, setId] = React.useState<NodeJS.Timer | undefined>(undefined);
```

**Review by matthiasblum — APPROVED** [review]
Looks good to me. I checked a few pages, didn't see anything wrong, and I made a few minor comments.

File: src/components/ProteinViewer/ConservationMockupTrack/index.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      </div>
+      <div className={css('track-label')}>
+        <button
+          type="button"
+          className={css('hollow', 'button', 'user-select-none')}
+          onClick={() => handleConservation()}
+        >
+          Fetch Conservation
+        </button>
+      </div>
```
[the remark this answers is not in the document]
Comment (gustavo-salazar) [reply src/components/ProteinViewer/ConservationMockupTrack/index.tsx:68]: As discussed in standup, this feature will be temporarily disabled until hmmer is working OK. A discussion on its redisign can be made as a story, when re-enabaling it.

File: src/components/ProteinViewer/Options/index.tsx [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+                );
+              });
+            });
+
+            str = str + fontCSS + colorsCSS + vfIproCSS;
+            style.innerHTML = str;
+            base.appendChild(style);
+          } else
+            console.warn(
+              "Couldn't setups the style for the protvista-saver snapshot "
```
[the remark this answers is not in the document]
Comment (gustavo-salazar) [reply src/components/ProteinViewer/Options/index.tsx:103]: change made in branch `other-sources-pv`

**Pull request merged**